qiskit/transpiler/transpile_circuit.py

```diff
@@ -22,61 +22,64 @@
 from qiskit.transpiler.exceptions import TranspilerError
 
 
-def transpile_circuit(circuit, transpile_config):
+def transpile_circuit(transpile_args):
```

Comment:
passing in a dict here (`transpile_args`) does not look like a good API. 

try expanding it:
`transpile_circuit(circuit, pass_manager, pass_manager_config, output_name, callback)`
and give them defaults

EDIT: i just saw @mtreinish made a similar comment.

Reply:
I'm returning back to the previous situation. Making that change would be changing the API...

Reply:
This is not the old API, it's now a dict rather than a TranspileConfig.

But that's not what I care about. My comment wasn't about "let's not change API". My point was "let's not have a hard-to-use API". I think a dict is hard to use. If we know the keys of the dict, might as well expand it into kwargs.

Reply:
This snippet is outdate. Check here https://github.com/Qiskit/qiskit-terra/pull/3035/files#diff-751f19949d593b00af720b607ee5f7e5R213 

I want to put it back, because is an irrelevant change. I'm thinking on deprecating this function in a coming PR.

Reply:
I'm referring to the current up to date version. You have changed TranspileConfig to a dict, so it is an API change. And I'm referring to this dict when I say this is a bad API.
```python
        transpile_config (dict): configuration dictating how to transpile.
            The format of the dictionary has the following format:
            {'circuit': QuantumCircuit,
             'optimization_level': int,
             'pass_manager': PassManager,
             'output_name': string,
             'callback': callable,
             'pass_manager_config': PassManagerConfig}
```

Reply:
I see. Making that change. Can I also deprecate `transpile_circuit`? Or that is for a different PR?

Reply:
Yeah I don't think we need to deprecate that.

Reply:
Can I remove `transpile_circuit` directly? Or are you talking about the parameters?